### PR TITLE
fix: maker checker condition handling

### LINKED ISSUE #1181: No pop up message shows 'Waiting for Checker approval"
1) Go to System>Configurations and enable Maker-Checker
2) Enable Create client checker in 'Configure Maker Checker tasks.
3) Go to left navigation page and tap on Clients.
4) In clients list page, click on '+' icon.
5) In next create client page, enter all mandatory fields and submit it. 

Expected: After Submission, should show message, 'Waiting for Checker approval' and it must be listed in checker Inbox. 

Actual: No pop up message shows 'Waiting for Checker approval;shows client created successfully. though it is listed in Checker Inbox.

### PR DESCRIPTION
**Pull request by laxyapahuja**
Fixes #1181 

### When Maker Checker is enabled:
ezgif-4-ed536e829d5d

### When Maker Checker is disabled: 
ezgif-4-46f643811a5e

- [x] Apply the `MifosStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by oorjitchowdhary** [comment]
this looks like an interesting issue fix, good work @laxyapahuja!

**Comment by laxyapahuja** [comment]
@moksh-mahajan @elirehema @luckyman20 can you please review this task? :D

**Comment by moksh-mahajan** [comment]
LGTM. Good Job @laxyapahuja

**Comment by laxyapahuja** [comment]
will this get merged?

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/datatablelistfragment/DataTableListPresenter.java [hunk 1]
```diff
@@ -121,7 +121,14 @@ public void onError(Throwable e) {
...
                         getMvpView().showProgressbar(false);
-                        getMvpView().showClientCreatedSuccessfully(client);
+                        if (client.getClientId() != null) {
+                            getMvpView().showClientCreatedSuccessfully(
+                                    client);
+                        } else {
+                            getMvpView().showWaitingForCheckerApproval(
+                                    R.string.waiting_for_checker_approval);
+                        }
+                        //getMvpView().showClientCreatedSuccessfully(client);
```
Comment (luckyman20) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/datatablelistfragment/DataTableListPresenter.java]: Is this line still needed?

Comment (laxyapahuja) [reply mifosng-android/src/main/java/com/mifos/mifosxdroid/online/datatablelistfragment/DataTableListPresenter.java]: updated :D

**Pull request merged**